### DISCUSSION TITLE
[Localization] Target all subdomains

### DISCUSSION THREAD

**Discussion by Netail** [vercel/next.js#34747] [Ideas]
### Describe the feature you'd like to request

Next i18n currently checks if the domain is **exactly** the same and only supports **1 domain per defaultLocale**.
So this makes it hard to include tst/acc/alpha/beta subdomains per domain.

For example, the domains `tst.google.nl` & `google.nl` should use nl-NL and `tst.google.be` & `google.be` should use nl-BE. 

#### Some examples

The specified defaultLocale doesn't work for the subdomains, so it uses the global defaultLocale
```
domains: [
    {
        domain: 'google.nl',
        defaultLocale: 'nl-NL',
    },
    {
        domain: 'google.be',
        defaultLocale: 'nl-BE',
    }
],
```

And this gives an build error as only 1 domain per defaultLocale is supported
```
domains: [
    {
        domain: 'tst.google.nl',
        defaultLocale: 'nl-NL',
    },
    {
        domain: 'google.nl',
        defaultLocale: 'nl-NL',
    },
    {
        domain: 'tst.google.be',
        defaultLocale: 'nl-BE',
    },
    {
        domain: 'google.be',
        defaultLocale: 'nl-BE',
    }
],
```

### Describe the solution you'd like

Regex support?
```
domains: [
    {
        domain: /google\.nl/,
        defaultLocale: 'nl-NL',
    },
    {
        domain: /google\.be/,
        defaultLocale: 'nl-BE',
    }
],
```

### Describe alternatives you've considered

Multiple domains of same defaultLocale
```
domains: [
    {
        domain: 'tst.google.nl',
        defaultLocale: 'nl-NL',
    },
    {
        domain: 'google.nl',
        defaultLocale: 'nl-NL',
    },
    {
        domain: 'tst.google.be',
        defaultLocale: 'nl-BE',
    },
    {
        domain: 'google.be',
        defaultLocale: 'nl-BE',
    }
],
```

**Comment by 09039598961** [comment]
Liked all hacking
